include/rxmesh/kernels/attribute.cuh

```diff
@@ -94,6 +94,86 @@ __launch_bounds__(blockSize) __global__
     }
 }
 
+struct CustomMaxPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value > a.value) ? b : a;
+    }
+};
+
+struct CustomMinPair
+{
+    template <typename T>
+    __device__ __forceinline__ cub::KeyValuePair<int, T> operator()(
+        const cub::KeyValuePair<int, T>& a,
+        const cub::KeyValuePair<int, T>& b) const
+    {
+        return (b.value < a.value) ? b : a;
+    }
+};
+template <class T, uint32_t blockSize, typename HandleT>
+__launch_bounds__(blockSize) __global__
+    void arg_max_kernel(const Attribute<T, HandleT> X,
+                    bool                        is_min,
+                    const uint32_t              num_patches,
+                    const uint32_t              num_attributes,
+                    cub::KeyValuePair<int, T>*  d_block_output,
+                    uint32_t                    attribute_id)
+{
+    using LocalT = typename HandleT::LocalT;
+
+    assert(X.get_num_attributes() == 1); //we can only take arg max for a scalar attribute
+
+    uint32_t p_id = blockIdx.x;
+    if (p_id < num_patches) {
+        const uint16_t element_per_patch = X.size(p_id);
+        cub::KeyValuePair<int, T> thread_val;
+        thread_val.value = std::numeric_limits<T>::lowest();
+        thread_val.key   = 0;
+        for (uint16_t i = threadIdx.x; i < element_per_patch; i += blockSize) {
+
+            if (X.get_patch_info(p_id).is_owned(LocalT(i)) &&
+                !X.get_patch_info(p_id).is_deleted(LocalT(i))) {
+
+                if (attribute_id != INVALID32 ) 
```

Comment:
In case `attribute_id` is `INVALID32`, we should treat the (vector) attribute as just a big flat array, i.e., return the vertex/edge/face handle of the entry with max/min value. This would just require a for loop over the number of attributes per element and apply the same min/max operation on it and store this in `thread_val`